src/dotnet/APIView/APIView/Model/V2/ReviewToken.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace APIView.Model.V2
+{
+    /// <summary>
+    /// Token corresponds to each component within a code line. A separate token is required for keyword, punctuation, type name, text etc.
+    /// </summary>
+    public class ReviewToken
+    {
+        public ReviewToken() { }
+        public ReviewToken(string value, TokenKind kind)
+        {
+            Value = value;
+            Kind = kind;
+        }
+        public TokenKind Kind { get; set; }
+        public string Value { get; set; }
+
+        /// <summary>
+        /// NavigationDisplayName property is to add a short name for the token that will be displayed in the navigation object.
+        /// </summary>
+        public string NavigationDisplayName { get; set; }
```

Comment:
Shouldn't this be a property of the ReviewLine?

Reply:
It can be at review line. I have included it here to make it easier for parser and also token contains class name. I don't prefer to duplicate class name property at line again.